File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.js
function solution(n) {
    var answer = 0;
//     재귀+메모이제이션
    let map = new Map()
    map.set(0,0n)
    map.set(1,1n)
    for(let i=2;i<=100000;i++){
        map.set(i,(map.get(i-1)+map.get(i-2)))
    
    }
    answer = (map.get(n)%1234567n)
    return Number(answer);
}

File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.py
import sys
sys.setrecursionlimit(10**6)

def solution(n):
    answer = 0
    dic = {}
    dic[0] = 0
    dic[1] = 1
    dic[2] = 1
    def fibo(num):
        if num in dic:
            return dic[num]
        else:
            dic[num] = fibo(num-1) + fibo(num-2)
            return dic[num]
    answer = fibo(n)
    return answer%1234567